Formula/allureofthestars.rb

```diff
@@ -0,0 +1,24 @@
+require "language/haskell"
+
+class Allureofthestars < Formula
+  include Language::Haskell::Cabal
+
+  desc "Near-future Sci-Fi roguelike and tactical squad combat game"
+  homepage "http://allureofthestars.com"
+  url "https://hackage.haskell.org/package/Allure-0.9.4.0/Allure-0.9.4.0.tar.gz"
+  sha256 "503cd08dd6dd71d0afe63920b8fa171047449e95a35369dab0936c490d3dabf4"
+  head "https://github.com/AllureOfTheStars/Allure.git"
+
+  depends_on "cabal-install" => :build
+  depends_on "ghc" => :build
+  depends_on "pkg-config" => :build
+  depends_on "sdl2_ttf"
+
+  def install
+    install_cabal_package :using => ["happy", "alex"]
+  end
+
+  test do
+    system "#{bin}/Allure", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thank you for the comment. I'd love to run my standard test, one that reads and parses a config file and font file, inits SDL, refrains from opening a window, but runs 50 frames of simulation without live output and prints a summary (and crashes outright if anything is out of order):

https://github.com/AllureOfTheStars/homebrew-allure/blob/97c5820379b8da19b391f4dd3d2148fec07a3221/HomebrewFormula/allureofthestars.rb#L22

However, apparently `sandbox-exec` blocks access to the fonts or breaks it otherwise, because the test invoked from the shell succeeds, but invoked from the formula, inside `sandbox-exec`, it fails:

https://travis-ci.org/AllureOfTheStars/Allure/jobs/522453118#L2507

Moreover, the formula seems to intercept the error message my program supposedly emits, even when I run the test with `--verbose`, so I can't debug properly. I must be doing something silly. Any hints? (I also tried to give the commandline arguments all on one line, but then it complains the line is too long.)

Reply:
Sandbox is a limitation, but a lot of things can still run. You can redirect output with `shell_output("… 2>&1")` for example. Running with `--verbose --debug` can also help.

Reply:
Fixed. The test is now quite exhaustive.